gcc/rust/parse/rust-parse-impl.h

```diff
@@ -12078,6 +12083,14 @@ Parser<ManagedTokenSource>::left_denotations (std::unique_ptr<AST::Expr> expr,
   while (right_binding_power < left_binding_power (current_token))
     {
       lexer.skip_token ();
+      if (restrictions.stop_on_token
+	  && (current_token->get_id () == MATCH_ARROW
+	      || current_token->get_id () == COMMA))
```

Comment:
this seems weird. you'd want to enter left denotation and create the infix with =>